### DISCUSSION TITLE
activity-aggregate-updater-job is failing.

### DISCUSSION THREAD

**Discussion by rupeshpyeole** [Sunbird-Lern/Community#159] [Issues] [answered]
Hi Team,
We are getting the error while running "activity-aggregate-updater-job " ,
for reference logs has been attached.
[activityaggregatorjobmanagerlogs.txt](https://github.com/Sunbird-Lern/Community/files/12670272/activityaggregatorjobmanagerlogs.txt)

we kindly request you to help us on resolving the issue.
cc: @manojkumarsah

Thanks,
Rupesh

**Comment by kumarks1122** [comment]
@rupeshpyeole : As per the log the job is failing because the collection `do_3138840147146342401947` is not having leafnodes details in the relation cache. Kindly check if the relation cache updater job up and running fine.
Also please check relation cache updater and activity aggregate updater flink jobs are having same redis configuration.

**Reply by Krishnaj20** [reply]
@rupeshpyeole @manojkumarsah - Is your query answered? If so, please close the discussion thread.

**Reply by shridhar95** [reply]
Hi @kumarks1122 @Krishnaj20,
The relation cache updater job is working fine, but still in redis we are unable to see the leafnodes for few do_id's. 
please suggest us how can we proceed on this.

for your reference please find the below errror message :
java.lang.Exception: leaf nodes are not available for a published collection: do_31388423575858380811211

**Reply by Jayaprakash8887** [reply]
@shridhar95 pls try pushing an event manually for the 'do_31388423575858380811211' to relation-cache-updater and see trough the logs if there is any exception

**Reply by Jayaprakash8887** [reply, marked as answer]
@shridhar95 Please let us know if the issue is resolved. If yes, kindly mark the ticket as answered
